hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -396,19 +404,33 @@ private ValidationResult validate(
                     DUPLICATE_TRANSACTION);
         }
 
+        // Check the status and solvency of the payer
+        try {
+            final var payer = solvencyPreCheck.getPayerAccount(storeFactory, payerID);
+            solvencyPreCheck.checkSolvency(txInfo, payer, fees.totalWithoutServiceFee());
+        } catch (final PreCheckException e) {
+            return new ValidationResult(NODE_DUE_DILIGENCE_FAILURE, e.responseCode());
+        }
+
         // Check the time box of the transaction
         try {
             checker.checkTimeBox(txBody, consensusNow);
         } catch (final PreCheckException e) {
             return new ValidationResult(PRE_HANDLE_FAILURE, e.responseCode());
         }
 
-        // Check the status and solvency of the payer
-        try {
-            final var payer = solvencyPreCheck.getPayerAccount(storeFactory, preHandleResult.payer());
-            solvencyPreCheck.checkSolvency(preHandleResult.txInfo(), payer, fees.totalWithoutServiceFee());
-        } catch (final PreCheckException e) {
-            return new ValidationResult(NODE_DUE_DILIGENCE_FAILURE, e.responseCode());
+        // Check if the payer has the required permissions
```

Comment:
The difference between 'has the required permissions' and 'has the required privileges' is not very clear. Maybe add a link to https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/docs/privileged-transactions.md in the javadoc.